embedded-graphics/src/mock_display.rs

```diff
@@ -238,6 +238,34 @@ where
         self.pixels[i as usize] = color;
     }
 
+    /// Get the affected area of the display.
+    pub fn affected_area(&self) -> Rectangle {
+        let (tl, br) = self
+            .pixels
+            .iter()
+            .enumerate()
+            .filter_map(|(idx, p)| p.map(|_| idx))
+            .fold(
+                (None, None),
+                |(tl, br): (Option<Point>, Option<Point>), idx| {
+                    let x = idx % SIZE;
+                    let y = idx / SIZE;
+
+                    let point = Point::new(x as i32, y as i32);
+
+                    (
+                        tl.map(|tl| tl.component_min(point)).or(Some(point)),
+                        br.map(|br| br.component_max(point)).or(Some(point)),
+                    )
+                },
+            );
+
+        Rectangle::with_corners(
+            tl.unwrap_or_else(Point::zero),
+            br.unwrap_or_else(Point::zero),
+        )
```

Comment:
This will not produce a zero sized rectangle if `tl` and `br` are `None`. I would also suggest to add a test for this case.